core/pkg/resourcehandler/remotegitutils.go

```diff
@@ -18,9 +38,26 @@ func cloneRepo(gitURL giturl.IGitURL) (string, error) {
 		return "", fmt.Errorf("failed to create temporary directory: %w", err)
 	}
 
-	// Clone option
+	// Get the URL to clone
 	cloneURL := gitURL.GetHttpCloneURL()
-	cloneOpts := git.CloneOptions{URL: cloneURL}
+
+	isGitRepoPublic := isGitRepoPublic(cloneURL)
+
+	// Declare the authentication variable required for cloneOptions
+	var auth transport.AuthMethod
+
+	if isGitRepoPublic {
```

Comment:
Is it necessary to pre-check if the repo is public or private? What if you pass an empty token?

Reply:
* If we are using `auth` in cloneOptions, it is throwing an error even for public repositories if the token is empty.
* In my understanding, we have to implement different authentication logic for public and private repositories. So, we should definitely pre-check if the repo is public or private. Correct me if I am wrong.